server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -0,0 +1,400 @@
+      return this;
+   }
+
+   public ReactorNettyJaxrsServer setClientAuth(final ClientAuth clientAuth)
+   {
+      Objects.requireNonNull(clientAuth);
+      this.clientAuth = clientAuth;
+      return this;
+   }
+
+   /**
+    * Sets clean up tasks that are needed immediately after {@link org.jboss.resteasy.spi.Dispatcher#invoke} yet before
+    * any asynchronous asynchronous work is continued by the reactor-netty server.  Since these run on the Netty event
+    * loop threads, it is important that they run fast (not block).  It is expected that you take special care with
+    * exceptions.  This is useful in certain cases where servlet Filters have options that are hard to achieve with the
+    * pure JAX-RS API, such as:
+    *{code}
+    *  doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) {
+    *     establishThreadLocals();
+    *     chain.doFilter(req, resp, chain);
+    *     clearThreadLocals();
+    *  }
+    *{code}
+    * @param cleanUpTasks List of clean up tasks
+    * @return ReactorNettyJaxrsServer
+    */
+   public ReactorNettyJaxrsServer setCleanUpTasks(final List<Runnable> cleanUpTasks) {
+      this.cleanUpTasks = cleanUpTasks;
+      return this;
+   }
+
+   private void configure(final Connection conn)
+   {
+      if (idleTimeout != null) {
+         final long idleNanos = idleTimeout.toNanos();
+         conn.channel().pipeline().addFirst("idleStateHandler", new IdleStateHandler(0, 0, idleNanos, TimeUnit.NANOSECONDS));
+         conn.channel().pipeline().addAfter("idleStateHandler", "idleEventHandler", new ChannelDuplexHandler() {
+            @Override
+            public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+               if (evt instanceof IdleStateEvent) {
+                  IdleStateEvent e = (IdleStateEvent) evt;
+                  if (e.state() == IdleState.ALL_IDLE) {
+                     ctx.close();
+                  }
+               }
+            }
+         });
+       }
+   }
+
+   private SslContext toNettySSLContext(final SSLContext sslContext)
+   {
+      Objects.requireNonNull(sslContext);
+      return new JdkSslContext(
+          sslContext,
+          false,
+          null,
+          IdentityCipherSuiteFilter.INSTANCE,
+          null,
+          clientAuth,
+          null,
+          false
+      );
+   }
+
+   private ResteasyUriInfo extractUriInfo(final HttpServerRequest req, final String contextPath)
+   {
+      final String uri = req.uri();
+
+      String uriString;
+
+      // If we appear to have an absolute URL, don't try to recreate it from the host and request line.
+      if (uri.startsWith(req.scheme() + "://")) {
+         uriString = uri;
+      } else {
+         uriString = new StringBuilder(100)
+             .append(req.scheme())
+             .append("://")
+             .append(req.hostAddress().getHostString())
+             .append(":").append(req.hostAddress().getPort())
```

Comment:
It looks like the `HttpServerRequest.hostAddress()` can be `null` so this has potential to produce an NPE.

Reply:
I will look into this.

Reply:
Javadoc says it will be null in the case of Unix Domain sockets.  I don't know much about those.  Do you think that is something we need to support or that the potential NPE should be handled better here?

I checked the netty4 adapter and it does seem susceptible to a similar NPE although maybe the caller guards there..  [code](https://github.com/resteasy/Resteasy/blob/4ded2d4f4e880fb50d1eaf88fc4af5e6e2bd5646/server-adapters/resteasy-netty4/src/main/java/org/jboss/resteasy/plugins/server/netty/NettyHttpRequest.java#L462).

Reply:
I have to admit I don't know much about it either. I don't really know what the URI would look like for that either. If we haven't seen a bug yet for Netty4 I say we can leave it as is :)